{"elemID":"component_293","date":"6/5/2020","time":"12:00am","name":"details","pathname":"details","pathstr":"async.during~details","path":[["/","async"],[".","during"],["~","details"]],"flags":[],"source":[["/","async"],[".","during"]],"sourcestr":"async.during","superClasses":[],"interfaces":[],"satisfies":[],"ctype":"spare","valtype":[],"isKwarg":false,"isMultiArg":false,"isFunction":false,"isSpare":true,"isModule":false,"isClass":false,"simpleCtype":"spare","hideCtype":false,"spares":[],"modules":[],"enums":[],"statics":[],"functions":[],"members":[],"methods":[],"localValues":[],"localFunctions":[],"arguments":[],"finalArgs":[],"finalKwargs":[],"returns":[],"throws":[],"names":[],"events":[],"signatures":[],"propertySymbols":[],"memberSymbols":[],"returnsSymbols":[],"hasChildren":false,"doc":[{"doc":"\n*\n     * Like [`whilst`]{@link module:ControlFlow.whilst}, except the `test` is an asynchronous function that\n     * is passed a callback in the form of `function (err, truth)`. If error is\n     * passed to `test` or `fn`, the main callback is immediately called with the\n     * value of the error.\n     *\n     * @name during\n     * @static\n     * @memberOf module:ControlFlow\n     * @method\n     * @see [async.whilst]{@link module:ControlFlow.whilst}\n     * @category Control Flow\n     * @param {Function} test - asynchronous truth test to perform before each\n     * execution of `fn`. Invoked with (callback).\n     * @param {Function} fn - A function which is called each time `test` passes.\n     * The function is passed a `callback(err)`, which must be called once it has\n     * completed with an optional `err` argument. Invoked with (callback).\n     * @param {Function} [callback] - A callback which is called after the test\n     * function has failed and repeated execution of `fn` has stopped. `callback`\n     * will be passed an error, if one occured, otherwise `null`.\n     * @example\n     *\n     * var count = 0;\n     *\n     * async.during(\n     *     function (callback) {\n     *         return callback(null, count < 5);\n     *     },\n     *     function (callback) {\n     *         count++;\n     *         setTimeout(callback, 1000);\n     *     },\n     *     function (err) {\n     *         // 5 seconds have passed\n     *     }\n     * );\n","context":[]}],"breadcrumbs":[{"path":[["/","async"]],"name":"async","delimiter":"/"},{"path":[["/","async"],[".","during"]],"name":"during","delimiter":"."},{"path":[["/","async"],[".","during"],["~","details"]],"name":"details","delimiter":"~"}],"baseTagPath":"../../../../../../"}